Realm/Tests/RealmTests.mm

```diff
@@ -2939,3 +2939,76 @@ - (void)testDeleteOpenRealmFile {
 }
 
 @end
+
+@interface InMemoryLogger : RLMLogger
+@property (nonatomic, strong) NSString *logs;
+@end
+
+@implementation InMemoryLogger
+- (instancetype)initWithTitle:(NSString *)title logLevel:(RLMLogLevel)logLevel {
+    self = [super init];
+    if (self) {
+        super.level = logLevel;
+        self.logs = title;
+    }
+    return self;
+}
+
+- (void)doLog:(RLMLogLevel)logLevel message:(NSString *)message {
+    if (!self.logs) {
+        self.logs = @"";
+    }
+    NSString *newLogs = [_logs stringByAppendingFormat:@" %@ %lu %@", [NSDate now], logLevel, message];
+    self.logs = newLogs;
+}
+@end
+
+@interface RLMLoggerTests : RLMTestCase
+@end
+
+@implementation RLMLoggerTests
+- (void)testCustomLogger {
+    // Default Log Level is RLMLogLevelOff
+    InMemoryLogger *inMemoryLogger = [[InMemoryLogger alloc] init];
+    [RLMLogger setDefaultLogger:inMemoryLogger];
+
+    @autoreleasepool { [RLMRealm defaultRealm]; }
+    XCTAssertTrue([inMemoryLogger.logs length] == 0);
+
+    // Test LogLevel Info
+    inMemoryLogger.level = RLMLogLevelInfo;
+    @autoreleasepool { [RLMRealm defaultRealm]; }
+    XCTAssertTrue([inMemoryLogger.logs length] >= 0);
+    XCTAssertTrue([inMemoryLogger.logs rangeOfString:@"4 DB:" options:NSRegularExpressionSearch].location != NSNotFound); // Info
```

Comment:
Why are these doing regex searches?

Reply:
Trying to check if the log for the specific log level are been added to the log string we are using for testing

Reply:
It's searching for a literal string though, not a regular expression. These can just be `containsString:` rather than a regex search.